Dockerfile

```
@@ -22,7 +22,8 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
         automake \
         autoconf \
         m4 \
-        gperf && \
+        gperf \
+        wget && \
     pip install ply && \
```

Non-blocker and a bit nitpicking, but don't you think that it's better if you move the RM to the layer where you run apt and move pip to a different layer? So you use cache more effectively and don't need to reinstall a lot of apt packages just because the ply package is updated.

```suggestion
    rm -rf /var/lib/apt/lists/*
RUN pip install ply
```

---

I don't think it really matters, I mean, current plans are to update the container... never? :) I mean, it's built once and then is used to run the build within and that's it.

Plus I don't want to upstream that...

---

@JustAMan I know this container will never be updated, as it's used to build the package. However, this doesn't mean that you can follow [better practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/) for this Dockerfile to speed up building time, as it doesn't use the Docker layers efficiently. I don't know if you're aware about this, but each of the instructions in a Dockerfile are run in different __layers__. When you build a container, each of those layers are cached, so when you build that same image again, redundant steps are omitted.

An instruction requires all the __layers__ that are above that instruction. When running ``docker build``, docker checks for each of the instruction if the resulting layer = cache. If not, runs the layer again, making all the layers below that level completely obsolete. That's why a proper sorting of the layers can make your build last from 1 minute (based in the amount of packages you're installing) to 2 seconds.

Imagine the following:

* You build the container today for the first time. Each of the layers is build, as there is no presence of them in cache.
* You make some changes to your code and run the build again. As all the above layers have been built before, Docker starts by recreating the layer that was changed (line 30-31 in your Dockerfile). The build will take **sensibly** less, because a huge step (getting dependencies) was already skipped.

Now, you go back to your code after a day. However, **ply** devs decided to update their package:

- **With your current implementation**: You would need to wait 1 minute for all the dependencies (__which none of them have changed, only ply__) to download & install dependencies + all the bottom layers.

- **Running pip in a different layer**: Docker would reuse all the apt packages and it will only recreate the layer that runs ``pip`` and all the bottom ones.

It might seem silly, but when you need to build multiple components (like we do), every second we can squeeze is good! Also, it speeds up testing while developing!

**Of course, if all the builds of this are done by invalidating cache running ``docker build --no-cache`` this is irrelevant.**

---

@JustAMan This is a great example of a Dockerfile by our fellow Joshua, as it runs the dependencies in one layer, yarn install in another one and build in another one, so everything is splitted in indivisible parts as much as possible: https://github.com/jellyfin/jellyfin-web/blob/5e00dd92051cce81352b06aad8fe9b5bb6a13d93/deployment/Dockerfile.portable

---

I know about layers and such, I was just feeling lazy, plus extracting `pip install` to a separate layer is completely separate from this PR target feature-wise (1 PR == 1 atomic feature, remember? and Dockerfile cleanup is a different feature). So I take this as a legitimate excuse to not fix it in this PR 😛

---

I'll give the explicit approval then, I thought it was worth commenting as you asked for specific review in the Dockerfile in Matrix :)

---

@ferferga I actually decided on a whim that I'm not feeling lazy, so there is ddf29596591053db6b8a354f7ba5bf61cb687b81 😄